packages/devtools_app/lib/src/debugger/program_explorer.dart

```diff
@@ -327,25 +327,29 @@ class _ProgramOutlineView extends StatelessWidget {
         if (isLoadingOutline) {
           return const CenteredCircularProgressIndicator();
         }
-        if (controller.outlineNodes.value == null ||
-            controller.outlineNodes.value.isEmpty) {
-          return const Center(
-            child: Text('Nothing to inspect'),
-          );
-        }
-        return TreeView<VMServiceObjectNode>(
-          itemExtent: _programExplorerRowHeight,
-          dataRootsListenable: controller.outlineNodes,
-          onItemSelected: onItemSelected,
-          onItemExpanded: onItemExpanded,
-          dataDisplayProvider: (node, onTap) {
-            return _ProgramExplorerRow(
-              controller: controller,
-              node: node,
-              onTap: () async {
-                await node.populateLocation();
-                controller.selectOutlineNode(node);
-                onTap();
+        return ValueListenableBuilder<List<VMServiceObjectNode>>(
+          valueListenable: controller.outlineNodes,
+          builder: (context, nodes, _) {
+            if (nodes == null || nodes.isEmpty) {
+              return const Center(
+                child: Text('Nothing to inspect'),
```

Comment:
Let's add an optional parameter to TreeView, `emptyTreeDisplayBuilder` or some better name if you can think of one that defaults to building `const SizedBox()`. Then use this value in the TreeView build method when the data is empty. I just took a look at the code, and I don't think controller.outlineNodes.value will ever be null, so we shouldn't have to worry about the null case.

Reply:
Done, added optional parameter